sql/core/src/test/scala/org/apache/spark/sql/execution/AlreadyPlannedSuite.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.test.SharedSparkSession
+
+class AlreadyPlannedSuite extends SparkPlanTest with SharedSparkSession {
+
+  import testImplicits._
+
+  test("simple execution") {
+    val df = spark.range(10)
+    val planned = AlreadyPlanned.dataFrame(spark, df.queryExecution.sparkPlan)
+
+    checkAnswer(planned, identity, df.toDF().collect())
+  }
+
+  test("planning on top won't work - filter") {
+    val df = spark.range(10)
+    val planned = AlreadyPlanned.dataFrame(spark, df.queryExecution.sparkPlan)
+
+    val e = intercept[AnalysisException] {
+      planned.where('id < 5)
```

Comment:
I'm a bit concerned about this. This is limiting what the fallbacked v1 source can do. Maybe it's fine as fallbacked v1 source is not a normal v1 source.

Reply:
yeah, you're right. This would actually prevent the source from adding modifications to the input dataFrame, e.g. attach hidden columns or such. Let me fix this